karapace/karapace.py

```diff
@@ -201,385 +69,5 @@ def r(body, content_type, status=200):
-                self.r({"id": schema_id}, content_type)
-
-            schema_versions = sorted(list(schemas))
-            # Go through these in version order
-            for version in schema_versions:
-                schema = subject_data["schemas"][version]
-                parsed_version_schema = avro.schema.Parse(schema["schema"])
-                if parsed_version_schema == new_schema:
-                    self.r({"id": schema["id"]}, content_type)
-                else:
-                    self.log.debug("schema: %s did not match with: %s", schema["schema"], new_schema.to_json())
-
-            compatibility = subject_data.get("compatibility", self.ksr.config["compatibility"])
-
-            # Run a compatibility check between on file schema(s) and the one being submitted now
-            # the check is either towards the latest one or against all previous ones in case of
-            # transitive mode
-            if compatibility in TRANSITIVE_MODES:
-                check_against = schema_versions
-            else:
-                check_against = [schema_versions[-1]]
-
-            for old_version in check_against:
-                old_schema = avro.schema.Parse(subject_data["schemas"][old_version]["schema"])
-                compat = Compatibility(old_schema, new_schema, compatibility=compatibility)
-                try:
-                    compat.check()
-                except IncompatibleSchema as ex:
-                    self.log.warning("Incompatible schema: %s", ex)
-                    self.r(
-                        body={
-                            "error_code": 409,
-                            "message": "Schema being registered is incompatible with an earlier schema"
-                        },
-                        content_type=content_type,
-                        status=409
-                    )
-
-            # We didn't find an existing schema and the schema is compatible so go and create one
-            schema_id = self.ksr.get_schema_id(new_schema)
-            version = max(self.ksr.subjects[subject]["schemas"]) + 1
-            self.log.info(
-                "Registering subject: %r, id: %r new version: %r with schema %r, schema_id: %r", subject, schema_id, version,
-                new_schema.to_json(), schema_id
-            )
-        self.send_schema_message(subject, new_schema.to_json(), schema_id, version, deleted=False)
-        self.r({"id": schema_id}, content_type)
-
-    async def write_new_schema_remote(self, subject, body, master_url, content_type):
-        self.log.info("Writing new schema to remote url: %r since we're not the master", master_url)
-        response = await self.http_request(
-            url="{}/subjects/{}/versions".format(master_url, subject), method="POST", json=body, timeout=60.0
-        )
-        self.r(body=response.body, content_type=content_type, status=response.status)
-
-
-def main():
```

Comment:
Our `setup.py` looks for an entry point in this file so moving this without changing it breaks the console scripts.

And this is somewhat related to a bigger point that do we want to have a single entrypoint for starting the application as it is now with `karapace config.json` and make it possible to turn on/off the schema and REST features in the config file or do we want have two different command line tools for this e.g. `karapace_schema schema_config.json` and `karapace_rest rest_config.json`? I've previously thought of the first option but what are your thoughts?

Reply:
It would be ideal if we could serve both from the same process if needed but run them independently. 

I.e. we can have more entrypoints than one but I'd also like to be able to turn things on and off based on configuration.

Reply:
As a user i usually prefer tools that deal with a minimal number of things , just because the liberty offered by the composition opportunities . That's a personal preference though , and in this case i can't think of any good reason to do it one way versus the other other than the fact that we'd probably end up with less test code when all the logic is provided under one service

Reply:
So the reason why we want both options is that it'll be superbly memory efficient to run them from the same process but like you, many people would prefer them separately so we can cater to both crowds.